programs/config_api/src/config_instruction.rs

```diff
@@ -1,27 +1,15 @@
 use crate::id;
-use crate::ConfigState;
-use bincode::serialize;
-use serde_derive::{Deserialize, Serialize};
-use solana_sdk::instruction::{AccountMeta, Instruction};
-use solana_sdk::pubkey::Pubkey;
-use solana_sdk::short_vec;
-use solana_sdk::system_instruction;
+use crate::{ConfigKeys, ConfigState};
+use solana_sdk::{
+    instruction::{AccountMeta, Instruction},
+    pubkey::Pubkey,
+    system_instruction,
+};
 
-/// A collection of keys to be stored in Config account data.
-#[derive(Debug, Default, Deserialize, Serialize)]
-pub struct ConfigKeys {
-    // Each key tuple comprises a unique `Pubkey` identifier,
-    // and `bool` whether that key is a signer of the data
-    #[serde(with = "short_vec")]
-    pub keys: Vec<(Pubkey, bool)>,
-}
-
-impl ConfigKeys {
-    pub fn serialized_size(keys: Vec<(Pubkey, bool)>) -> usize {
-        serialize(&ConfigKeys { keys })
-            .unwrap_or_else(|_| vec![])
-            .len()
-    }
+fn initialize_account<T: ConfigState>(config_pubkey: &Pubkey) -> Instruction {
+    let account_metas = vec![AccountMeta::new(*config_pubkey, true)];
+    let account_data = (ConfigKeys { keys: vec![] }, T::default());
```

Comment:
@mvines @CriesofCarrots this line is suspect for me, I think we're basically equivalent to before, but perhaps we should actually be putting the keys in the account at create()+init() time.

the reason I didn't just use create()+store() was that a fair few places where create() was being called weren't signing according to the metas produced by they store() logic...

Reply:
Example?
Not certain I follow